### DISCUSSION TITLE
lots of new version!

### DISCUSSION THREAD

**Discussion by Stibax** [lostallmymoney/Razer_Mouse_Linux#29] [General]
Hi lostallmymoney,
i've seen that in these days you have made lots of new version of this fantastic program! I would like to know, if you have time, to show what new features you add? Thanks again for your great work.

**Comment by lostallmymoney** [comment]
Right now I've made the naga config owned by root to be more secure, added naga edit command to edit with ease, naga fix to restart usb services in case you were to restart the daemon once too many times, and made the overall code cleaner, also added an auto window config switcher, also made the logs go where you run "naga start" and overall reworked the scripts and code to be cleaner. I think 9.32 will be last version for today.
Also added specialKey to handle pressOnPress and pressOnRelease.

To do :
The custom fakekey library handles writing nicely but I'm looking if I can get it to execute media keys for when X11 starts breaking with xdotool, literally had to do a fresh install the other day >< (X11 freezes when you try to do xdotool keys).
Also trying to find the right fit to press keys on wayland, hence why I renamed the files to X11.
The best I found was dotool but it was lacking some special characters. Seems like a perfect solution is unfound as of right now.

I'll also be updating the readme in some time so you'll see all of it !

Happy to know you enjoy !

**Reply by lostallmymoney** [reply]
Also added a line to reload udev rules after installing so no need to restart.

**Reply by lostallmymoney** [reply]
And in this new service update I've made a new user that runs the service in prevision for when I find a solution to write events to uinput and be more secure that way no other application benefits from the udev rules.
Also starts with the OS

**Reply by lostallmymoney** [reply]
hold up i need to setup that xhost

**Reply by lostallmymoney** [reply]
Here so the fix was to add a line to the sudoer file to allow razerInput to use the user's account. I also made it so the $PATH is the same as the user's.

**Reply by lostallmymoney** [reply]
So finally ended rewriting it as it's only a group that's running the service to get udev ahah ! Also a little bit of toying to make sure the service is always up after logging out and back in. Anyways, if someone wants to give me any feedback on how it's working right now on their end it'd be great. (especially the auto start part)
With a service running inside the group if we asses that anything placed in the config will run with the group so might just be able to write to read and write just as well onto the keyboard, the normal processes launched by a normal click should not be able to.

**Comment by lostallmymoney** [comment]
Hey ! I'm wondering if you had any time to test out new features on your end ?

**Reply by Stibax** [reply]
Hi,
sure in the next few days i will test it and report back!

**Comment by Stibax** [comment]
Hi lostallmymoney, today i had time to upgrade to the latest version. I had  a very old version (before those massive update), so i decide to uninstall and reinstall the new version. All went smooth, i have only see an error regardiing a file: bash.aliases or somenthing like this. I didn't use bash but i use fish instead so i have to manualy make the alias naga=nagaX11. I don't use wayland so a cannot say nothing. The systemd units works well, even after suspend. I din't use yet complicated command, so i cannot say nothing. I use only run program e simpli combination. Thanks again for your work.

**Comment by lostallmymoney** [comment]
Very great.
The bash alias only indicates that you will need to use nagaX11 or nagaWayland commands instead of the naga alias I tried to setup :)
Thanks a lot !
________________________________
